BoxContentSDK/BoxContentSDKTests/BOXFileRepresentationDownloadRequestTests.m

```diff
@@ -0,0 +1,133 @@
+    
+    XCTestExpectation *expectation = [self expectationWithDescription:@"expectation"];
+    [request performRequestWithProgress:nil completion:^(NSError *error) {
+        XCTAssertNil(error);
+        [expectation fulfill];
+    }];
+    [self waitForExpectationsWithTimeout:2.0 handler:nil];
+    
+    NSData *data = [NSData dataWithContentsOfFile:localFilePath];
+    XCTAssertEqualObjects(cannedResponseData, data);
+}
+
+- (void)test_that_operation_is_not_marked_as_small_download
+{
+    NSString *fileID = @"123";
+    BOXRepresentation *rep = [self testRepresentation];
+    BOXFileRepresentationDownloadRequest *request = [[BOXFileRepresentationDownloadRequest alloc] initWithLocalDestination:@"/dummy/path" fileID:fileID representation:rep];
+    
+    BOXAPIOperation *operation = [request operation];
+    XCTAssert([operation isKindOfClass:[BOXAPIDataOperation class]]);
+    BOXAPIDataOperation *dataOperation = (BOXAPIDataOperation *)operation;
+    XCTAssert(dataOperation.isSmallDownloadOperation == NO);
+}
+
+#pragma mark - Error Handling
+
+- (void)test_that_invalid_grant_400_error_triggers_logout_notification
+{
+    NSString *fileID = @"123";
+    BOXRepresentation *rep = [self testRepresentation];
+    BOXFileRepresentationDownloadRequest *request = [[BOXFileRepresentationDownloadRequest alloc] initWithLocalDestination:@"/dummy/path" fileID:fileID representation:rep];
+    
+    NSData *cannedResponseData = [self cannedResponseDataWithName:@"invalid_grant"];
+    NSHTTPURLResponse *URLResponse = [self cannedURLResponseWithStatusCode:400 responseData:cannedResponseData];
+    [self setCannedURLResponse:URLResponse cannedResponseData:cannedResponseData forRequest:request];
+    
+    [request performRequestWithProgress:nil completion:^(NSError *error) {
+    }];
+    
+    [self expectationForNotification:BOXUserWasLoggedOutDueToErrorNotification object:nil handler:nil];
+    [self waitForExpectationsWithTimeout:2.0 handler:nil];
+}
+
+- (void)test_that_unauthorized_401_error_triggers_logout_notification
+{
+    NSString *fileID = @"123";
+    BOXRepresentation *rep = [self testRepresentation];
+    BOXFileRepresentationDownloadRequest *request = [[BOXFileRepresentationDownloadRequest alloc] initWithLocalDestination:@"/dummy/path" fileID:fileID representation:rep];
+    
+    NSHTTPURLResponse *URLResponse = [self cannedURLResponseWithStatusCode:401 responseData:nil];
+    [self setCannedURLResponse:URLResponse cannedResponseData:nil forRequest:request];
+    
+    [request performRequestWithProgress:nil completion:^(NSError *error) {
+    }];
+    
+    [self expectationForNotification:BOXUserWasLoggedOutDueToErrorNotification object:nil handler:nil];
+    [self waitForExpectationsWithTimeout:2.0 handler:nil];
+}
+
+# pragma mark - private helpers
+
+- (NSData *)randomDataWithLength:(NSUInteger)length
+{
+    NSMutableData *mutableData = [NSMutableData dataWithCapacity: length];
+    for (unsigned int i = 0; i < length; i++) {
```

Comment:
Minor: Probably you can move by `sizeof(NSInteger)` and reduce to 25% or less of the runtime.